### DISCUSSION TITLE
[Feature] Mobile App: Sharing images should share a link like the web version does

### DISCUSSION THREAD

**Discussion by phudgee** [immich-app/immich#1863] [Feature Request]
### The feature

On the web, I can select 20 images and share them. It creates a link to share. On the mobile app, I select 20 images and try to share them, it tries to send all 20 images. Would be nice if it just generated a link like the web version does. Thanks! Love this project! Been waiting for something like this for a long time!

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by Sherlock1979** [comment]
I think that in the mobile app we need both functionality. Share to other apps and share link.

**Reply by kamzil** [reply]
Exactly. Also, just offline preview photos should be enough when sharing to other apps. See my use case [here](https://github.com/immich-app/immich/discussions/1863#discussioncomment-8456580).

**Comment by bartowski1182** [comment]
Large agree on this feature, it's a big pain point, my friends want to use just the app and not the desktop so sharing public link is a must have for mobile

**Comment by schmitzkr** [comment]
What if I only need to share one image from the mobile app?  That probably shouldn't create an album/link right?

**Reply by bartowski1182** [reply]
I imagine you'd just share it like a normal photo in that case, not using the link share option

**Reply by schmitzkr** [reply]
> I imagine you'd just share it like a normal photo in that case, not using the link share option

Ah ok, yes I agree.  So this request is for an entirely new sharing feature in mobile, not to change the existing share function.

**Reply by darsor** [reply]
It should also be possible to share a single photo as a link. For example, sharing a photo directly (not a link) over SMS can severely degrade the photo/video quality, whereas sending a link to the photo via SMS allows the end user to access the original quality assets.

**Comment by NicholasFlamy** [comment]
If you go to an individual item and hit share (not share to) you get link sharing and normal albums have share which give you a link.

**Reply by AlanMW** [reply]
Yep, the option is available when you hold and select an image. So I guess the real ask here is that both the "Share" and "Share To" options be visible when viewing an image full screen.

**Comment by kamzil** [comment]
My use case is that sometimes I don't have a connection to my Immich server but I have downloaded the photo previews on my "offline" instance of the Immich on my Android phone. I would like to still be able to share the photo version that I have (preview in this case) to other apps on my phone, e.g. for editing or just send it to someone (it will be downscaled by messaging apps anyway so preview version is perfect for this). I don't see why it should be necessary to download the original one from the server for this, especially if in settings I've selected to only download the photo previews to my device.

**Reply by NicholasFlamy** [reply]
As a hidden advanced feature this could be a maybe. But it should never ever share the preview by default under any circumstances except with the setting explicitly enabled.

And if it's a hidden advanced feature then it's low priority for the devs.

**Reply by kamzil** [reply]
I agree that this shouldn't be the default, but disagree on this being an "advanced" feature or that it should be hidden. It could very well be given as a choice when sharing.

**Reply by NicholasFlamy** [reply]
By that, I meant the only way you would know about the feature is if you checked the docs or release notes or you dug into the app settings.

What you wouldn't want as the default out-of-box: Clicking share makes a popup asking whether to share the original or preview, adding an unnecessary step for 99% of users. (I am saying this even as I am in the 1%, before you commented, I made [#6723](https://github.com/immich-app/immich/discussions/6723#discussioncomment-8284870) which considers exactly what you stated. If you want this feature, upvote that post.)

**Reply by kamzil** [reply]
Yeah, we of course want the UX to be as smooth as possible. Glad to see that initiative!

**Comment by JeffreyvanNijnatten** [comment]
It is also confusing from an UI perspective.

When you (hold) select a photo you get the following options:
_-Share: You get the basic share icon with the line and 3 dots. But here it is the function to share a link.
-Share To: You get a box with an arrow to share with other apps_

When you immerse into a single photo:
_-Share icon: you get the basic share icon like before, but this time the function "Share To" is executed._


**Solutions:**

In the overview sections have the following options;
_Share Link: hyperlink icon
Share To: Share icon_

In the immersed view:
_Either add an extra button with the 'hyperlink icon' and allow it to share a link._

Or

_When you hit the 'share' button, allow the user the option to "Share Link"._

**Comment by NicholasFlamy** [comment]
Multiple comments here addressed the Photo Viewer so I have created a discussion for that: #15434

Since the original post was focused on link sharing in the Multi-Select View, which has been implemented, I am closing this discussion.
